tensorflow/core/ops/image_ops.cc

```diff
@@ -131,6 +131,56 @@ Status NMSShapeFn(InferenceContext* c) {
   return Status::OK();
 }
 
+Status NMSLiteShapeFn(InferenceContext* c) {
+  //Get inputs and validate ranks
+  ShapeHandle boxes;
+  //boxes is a tensor of Dimensions [batch_size, num_anchors, q, 4]
+  TF_RETURN_IF_ERROR(c->WithRank(c->input(0), 4, &boxes));
+  ShapeHandle scores;
+  //scores is a tensor of Dimensions [batch_size, num_anchors, num_classes]
+  TF_RETURN_IF_ERROR(c->WithRank(c->input(1), 3, &scores));
+  ShapeHandle max_output_size_per_class;
+  TF_RETURN_IF_ERROR(c->WithRank(c->input(2), 0, &max_output_size_per_class));
+  ShapeHandle max_total_size;
+  TF_RETURN_IF_ERROR(c->WithRank(c->input(3), 0, &max_total_size));
+  ShapeHandle iou_threshold;
+  TF_RETURN_IF_ERROR(c->WithRank(c->input(4), 0, &iou_threshold));
+  ShapeHandle score_threshold;
+  TF_RETURN_IF_ERROR(c->WithRank(c->input(5), 0, &score_threshold));
+
+  DimensionHandle unused;
+  //boxes[1] and scores[1] are both num_anchors
+  TF_RETURN_IF_ERROR(c->Merge(c->Dim(boxes, 1), c->Dim(scores, 1), &unused));
+  // The boxes[3] is 4.
+  TF_RETURN_IF_ERROR(c->WithValue(c->Dim(boxes, 3), 4, &unused));
+
+  bool pad_per_class;
+  TF_RETURN_IF_ERROR(c->GetAttr("pad_per_class", &pad_per_class));
+  if(!pad_per_class) {
+    DimensionHandle output_dim;
+    DimensionHandle batch_dim = c->Dim(boxes, 0);
+    
+    TF_RETURN_IF_ERROR(c->MakeDimForScalarInput(3, &output_dim));
+    if(c->ValueKnown(output_dim) && c->Value(output_dim) <= 0) {
+      return errors::InvalidArgument("max_total_size should be > 0 ");
+    }
+    c->set_output(0, c->MakeShape({batch_dim, output_dim, 4}));
+    c->set_output(1, c->MakeShape({batch_dim, output_dim}));
+    c->set_output(2, c->MakeShape({batch_dim, output_dim}));
+    c->set_output(4, c->MakeShape({batch_dim, output_dim}));
+  }
+  else {
+    //TODO SR Can all the outputs be unknowndim?
+    c->set_output(0, c->Vector(c->UnknownDim()));
+    c->set_output(1, c->Vector(c->UnknownDim()));
```

Comment:
we do know the output size:

min(max_output_size, max_output_size_per_class*classes)